### PR TITLE
Fix incorrect pos for intersection type definition

### LINKED ISSUE #40382: [Bug]:  Invalid location for Intersection types
### Description

sample source 
```
import ballerina/graphql;

public type Address readonly & record {
    int number;
    string street;
    string city;
};

isolated service on new graphql:Listener(9000) {

    isolated resource function get name() returns Address {
        return {number: 1, street: "", city: ""};
    }
}

```
when we try to retrieve location data of the Address intersection `TypeDefinitionSymbol`, an invalid position is provided as given below
image

proper location is received when we try the record without the 'readonly &'
### Steps to Reproduce

_No response_

### Affected Version(s)

2201.5.0

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes #40382

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public Symbol definition() {
             this.definition = symbolFactory.getBCompiledSymbol(((BParameterizedType) this.tSymbol.type).paramSymbol,
                                                                this.name());
         } else if (referredType.tag == TypeTags.INTERSECTION) {
-            this.definition = symbolFactory.getBCompiledSymbol(referredType.tsymbol,
+            this.definition = symbolFactory.getBCompiledSymbol(bType.tsymbol,
                     referredType.tsymbol.getName().getValue());
```
Comment (dulajdilshan) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java:108]: What is the reason for this?

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java:108]: Shouldn't we take the referred-type

Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java:108]: We should pass the `tsymbol` of the ref type right? If we do `referredType.tsymbol` we r passing the `tsymbol` of the intersection type

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java [hunk 2]
```diff
@@ -390,4 +391,21 @@ public void testHasDefaultValue(int line, int col, Boolean hasDefault) {
...
+        Project project = BCompileUtil.loadProject("test-src/symbol_position_bir_test.bal");
+        Package currentPackage = project.currentPackage();
+        ModuleId defaultModuleId = currentPackage.getDefaultModule().moduleId();
+        Document srcFile = getDocumentForSingleSource(project);
+
+        PackageCompilation packageCompilation = currentPackage.getCompilation();
+        SemanticModel model = packageCompilation.getSemanticModel(defaultModuleId);
+        LineRange lineRange = ((TypeReferenceTypeSymbol) model.symbol(srcFile,
+                from(19, 16)).get()).definition().getLocation().get().lineRange();
+        assertEquals(lineRange.startLine().line(), 143);
```
Comment (ushirask) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java:406]: Shall we keep `143` as a variable and have line 408 as that var + `4` to make it easier for future changes to the file?

Comment (dulajdilshan) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java:406]: mm. Good suggestion, but I think we don't need to do it in that way, Most of the Semantic API test cases in a way we compare exact values.

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java (hunk 1 above)
Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java:108]: Had a chat with @rdulmina live and we decided to go with this

File: tests/ballerina-compiler-api-test/src/test/resources/test-src/symbol_position_bir_test.bal [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (dulajdilshan) [inline tests/ballerina-compiler-api-test/src/test/resources/test-src/symbol_position_bir_test.bal:3]:
```suggestion
// WSO2 LLC. licenses this file to you under the Apache License,
```

**Review by dulajdilshan — COMMENTED** [review]
Only a minor change

File: tests/ballerina-compiler-api-test/src/test/resources/test-src/symbol_position_bir_test.bal (hunk 3 above)
Comment (rdulmina) [reply tests/ballerina-compiler-api-test/src/test/resources/test-src/symbol_position_bir_test.bal]: Fixed

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Pull request merged**